rmw_opensplice_cpp/src/functions.cpp

```diff
@@ -1034,53 +1039,80 @@ rmw_destroy_subscription(rmw_node_t * node, rmw_subscription_t * subscription)
 rmw_ret_t
 rmw_take(const rmw_subscription_t * subscription, void * ros_message, bool * taken)
 {
+  auto publisher_gid = rmw_take_with_gid(subscription, ros_message, taken);
+  bool result = (publisher_gid != nullptr);
+  rmw_destroy_gid(publisher_gid);
```

Comment:
I don't like the idea that every `rmw_take` creates a gid and destroys it immediately again.